### PR TITLE
Alert notification API

### PR DESCRIPTION
**Pull request by nicholaspcr**
#### Summary

References https://github.com/TheThingsIndustries/lorawan-stack/pull/3911
References https://github.com/TheThingsIndustries/product-management/issues/4

Introduces the documentation of Alert Notification API present on IS.

#### Screenshots

<details>
<summary> screenshots </summary>
<details>
<summary> Gateway.md update </summary>

image

</details>

<details>
<summary> IS API </summary>

localhost_1313_reference_api_identity_server_

image

</details>

<details>
<summary> profile CLI </summary>

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-profiles_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-profiles_create_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-profiles_delete_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-profiles_get_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-profiles_get-default_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-profiles_list_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-profiles_update_

</details>

<details>
<summary> Receivers CLI </summary>

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-receivers_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-receivers_create_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-receivers_delete_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-receivers_get_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-receivers_list_

localhost_1313_ttn-lw-cli_ttn-lw-cli_alert-notification-receivers_update_

</details>
</details>

#### Changes

-  Add the new generated doc files from ttes.
- Update API definition for IS
- Add `AlertNotificationProfileIdentifiers` message to gateway's API section.
- Update `ttn-lw-cli.json` to include new experimental CLI commands on the command line interface documentation section.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

**Comment by KrishnaIyer** [comment]
I think this PR should either go also to a feature branch or be held until the feature is complete and this PR covers the full functionality.

**Comment by nicholaspcr** [comment]
I agree, left it as a draft with the intention of only merging it when the PR referenced goes into a release.

Will do the same process as what is done on enterprise, change the target to a feature branch.

**Comment by nicholaspcr** [comment]
With the previous topic more defined, I'll make this PR ready for review after the `api` section of the PR is approved.
Avoiding duplicate work every time an API change is requested

**Comment by nicholaspcr** [comment]
@nejraselimovic could you give me a hand with an error I'm having on these changes?

After changing `doc/data/commands/ttn-lw-cli.json` I'm getting a bunch of `REF_NOT_FOUND` errors in CLI command files, which are not related to the new commands, and I can't identify what am I'm doing wrong.

Full error example:
```
ERROR 2023/11/02 19:27:14 [en] REF_NOT_FOUND: Ref "ttn-lw-cli/ttn-lw-cli_alert-notification-receivers_get" from page "ttn-lw-cli/ttn-lw-cli_users_set.md": page not found
```

**Comment by nejraselimovic** [comment]
> @nejraselimovic could you give me a hand with an error I'm having on these changes?
> 
> After changing `doc/data/commands/ttn-lw-cli.json` I'm getting a bunch of `REF_NOT_FOUND` errors in CLI command files, which are not related to the new commands, and I can't identify what am I'm doing wrong.
> 
> Full error example:
> 
> ```
> ERROR 2023/11/02 19:27:14 [en] REF_NOT_FOUND: Ref "ttn-lw-cli/ttn-lw-cli_alert-notification-receivers_get" from page "ttn-lw-cli/ttn-lw-cli_users_set.md": page not found
> ```

`/doc/data/commands` is being auto-generated with every new release, so I think you shouldn't be adding that manually? [See release docs template](https://github.com/TheThingsIndustries/lorawan-stack-docs/blob/master/.github/ISSUE_TEMPLATE/release.md)

File: doc/data/commands/ttn-lw-cli.json
Comment (nejraselimovic) [inline doc/data/commands/ttn-lw-cli.json:8]:
```suggestion
        "path": "ttn-lw-cli alert_notification_profiles",
```

Comment (nejraselimovic) [reply doc/data/commands/ttn-lw-cli.json:8]: I changed path to this and local build was successful, can you try that?

Comment (nicholaspcr) [reply doc/data/commands/ttn-lw-cli.json:8]: This was also my mistake, in the path there were two distinct files: one with `alert-notification-profiles` (generated by the commands you gave and another `alert_notifications_profiles` which I created. 

Everything is working now, thanks for the help

**Comment by nicholaspcr** [comment]
>/doc/data/commands is being auto-generated with every new release, so I think you shouldn't be adding that manually? [See release docs template](https://github.com/TheThingsIndustries/lorawan-stack-docs/blob/master/.github/ISSUE_TEMPLATE/release.md)

@nejraselimovic thanks, missed this somehow and was a bit lost. 
Silly mistake on my end and I appreciate you taking your time to help me.

**Review by nejraselimovic — APPROVED** [review]
To me it looks good, docs are building correctly, but I suggest you wait for another reviewers' approval just in case I'm missing something

**Review by adriansmares — APPROVED** [review]
LGTM.

I think some of the changes here are part of https://github.com/TheThingsIndustries/lorawan-stack-docs/pull/1206 , which wasn't available when this branch was created. I suggest rebasing on top of `master` both the feature branch and this branch.

**Pull request merged**
